test/extended/dr/machine_recover.go

```diff
@@ -0,0 +1,276 @@
+				newMaster.SetResourceVersion("")
+				newMaster.SetSelfLink("")
+				newMaster.SetUID("")
+				newMaster.SetCreationTimestamp(metav1.NewTime(time.Time{}))
+				// retry until the machine gets created
+				err = wait.PollImmediate(5*time.Second, 10*time.Minute, func() (bool, error) {
+					_, err := ms.Create(newMaster, metav1.CreateOptions{})
+					if errors.IsAlreadyExists(err) {
+						framework.Logf("Waiting for old machine object %s to be deleted so we can create a new one", replacedMaster.Name)
+						return false, nil
+					}
+					if err != nil {
+						return false, err
+					}
+					return true, nil
+				})
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				framework.Logf("Wait for masters to join as nodes and go ready")
+				err = wait.Poll(30*time.Second, 30*time.Minute, func() (done bool, err error) {
+					defer func() {
+						if r := recover(); r != nil {
+							fmt.Println("Recovered from panic", r)
+						}
+					}()
+					nodes, err := oc.AdminKubeClient().CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/master="})
+					if err != nil {
+						return false, err
+					}
+					ready := countReady(nodes.Items)
+					if ready != len(masters) {
+						framework.Logf("%d master nodes still unready", len(masters)-ready)
+						return false, nil
+					}
+					return true, nil
+				})
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				framework.Logf("Verify recreated master has healthy etcd endpoint")
+				healthyEtcds = getHealthyEtcds(certsDir, masters)
+				o.Expect(len(healthyEtcds)).To(o.BeNumerically("==", len(masters)))
+
+			})
+	},
+	)
+})
+
+func getHealthyEtcds(certsDir string, masters []*corev1.Node) []*etcdserverpb.Member {
+	endpoints := getEtcdEndpoints(masters)
+	etcdClient, err := getEtcdClient(certsDir, endpoints)
+	o.Expect(err).ToNot(o.HaveOccurred())
+	memberListResp, err := etcdClient.MemberList(context.Background())
+	o.Expect(err).ToNot(o.HaveOccurred())
+
+	healthyMembers := []*etcdserverpb.Member{}
+
+	for _, m := range memberListResp.Members {
+		_, err := etcdClient.Status(context.Background(), m.ClientURLs[0])
+		if err != nil {
+			healthyMembers = append(healthyMembers, m)
+		}
+	}
+	return healthyMembers
+}
+
+func getEtcdClient(certsDir string, endpoints []string) (*clientv3.Client, error) {
+	dialOptions := []grpc.DialOption{
+		grpc.WithBlock(), // block until the underlying connection is up
+	}
+
+	tlsInfo := transport.TLSInfo{
+		CertFile:      path.Join(certsDir, "tls.crt"),
+		KeyFile:       path.Join(certsDir, "tls.key"),
+		TrustedCAFile: path.Join(certsDir, "ca-bundle.crt"),
+	}
+	tlsConfig, err := tlsInfo.ClientConfig()
+
+	cfg := &clientv3.Config{
+		DialOptions: dialOptions,
+		Endpoints:   endpoints,
+		DialTimeout: 15 * time.Second,
+		TLS:         tlsConfig,
+	}
+
+	cli, err := clientv3.New(*cfg)
+	if err != nil {
+		return nil, err
+	}
+	return cli, err
+}
+
+func dumpEtcdCertsOnDisk(oc *exutil.CLI, dir string) {
```

Comment:
is there an easier way to get the certs on disk? the operator pod mounts the confimap and secret, is that option available for where this test is run?